File: scripts/supabase_db_test.js
require("dotenv").config();
const { createClient } = require("@supabase/supabase-js");

function requireEnv(name) {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Missing environment variable: ${name}`);
  }
  return value;
}

async function main() {
  const url = requireEnv("SUPABASE_URL");
  const key = requireEnv("SUPABASE_KEY");

  const supabase = createClient(url, key);
  const { data, error } = await supabase.from("chat_messages").select("*").limit(5);
  if (error) {
    throw error;
  }
  console.log(data ?? []);
}

main().catch((err) => {
  console.error(err);
  process.exitCode = 1;
});

File: scripts/supabase_db_test.py
"""Quick connectivity test against Supabase chat tables."""

import os

from dotenv import load_dotenv
from supabase import Client, create_client


def require_env(var_name: str) -> str:
    """Fetch env var or exit with clear message."""
    value = os.getenv(var_name)
    if not value:
        raise SystemExit(f"Missing environment variable: {var_name}")
    return value


def main() -> None:
    load_dotenv()

    url = require_env("SUPABASE_URL")
    anon_key = require_env("SUPABASE_KEY")

    client: Client = create_client(url, anon_key)
    response = client.table("chat_messages").select("*").limit(5).execute()
    print(response.data or [])


if __name__ == "__main__":
    main()
